### DISCUSSION TITLE
Unable to complete the "getting started with AI deck" tutorial

### DISCUSSION THREAD

**Discussion by abyvinodm** [bitcraze/discussions#216] [Q&A] [answered]
Hi,

I am have facing a roadblock when following the steps in 
https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/.

System: Ubuntu 20.04
Hardware: Crazyflie 2.1, AI Deck 1.1 (I can see Rev C on the board)
CFClient: 2022.05

I was able to follow the first step (update-crazyflie-and-aideck-firmware), and confirm `ESP32: I (910) SYS: Initialized`.

**Roadblock 1:** Seeing Rev C on my board, I assumed I had to do the steps in Gap8 bootloader (for older AIdecks revisions A-C). However, I repeatedly got stuck at `Initialising GAP8 JTAG TAP`. Can you help me with this?

**Roadblock 2:** I also took another route. I went back to the first step, skipped the Gap8 bootloader step, and moved on to the Flash Wifi Example. In this case, I got stuck at `\ 99% Writing to bcAI:gap8 deck memory`.

I tried all of these steps with another AI deck as well, and got stuck at the same points.

Any advice regarding this is much appreciated!

**Comment by knmcguire** [comment]
Hi!

As you noticed, I transferred the issue to a discussion as this is a support question.

If you have a REV C aideck and you got the AIdeck 1.1 directly from our store, then you we first need to fix roadblock  1 first or else roadblock 2 will never work. So let's focus on that.

- Could you copy past the entire terminal output when you are trying to flash the aideck. 
- Also which programmer are you using?

**Reply by knmcguire** [reply]
Also [the tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) has the following note:
`If flashing hangs at Initialising GAP8 JTAG TAP, you probably have to disconnect and reconnect the programmer and restart the Crazyflie.`

**Reply by abyvinodm** [reply]
Yes, that helped.

In case anyone else is wondering, I am using OLIMEX's ARM-JTAG-20-10 and ARM-USB-TINY-H.

**Comment by abyvinodm** [comment, marked as answer]
Hi!

A new try on a new day fixed it! :tada: (Not really sure how!) On fixing the roadblock 1, I was able to sail past roadblock 2 (as you correctly predicted!) Thank you!

Now, I am doing the last step. The tutorial says:

> In your wifi list you should see ‘WiFi streaming example’. If you do, please connect to it.
> Clone the [AIdeck example repository](https://github.com/bitcraze/aideck-gap8-examples) and select this WiFi and run [this script](https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples/other/wifi-img-streamer).
 
 I was able to see ‘WiFi streaming example’ in my wifi list, and connect to it successfully.
**Question**: What is the script I should be running? The link in the tutorial just points to a folder.

I went ahead and ignored the above step, and ran the python script in a separate environment with opencv-python installed. But, I got the following error:
![image](https://user-images.githubusercontent.com/72518986/185949292-09ecf268-8b8b-4c33-ba48-de5fd293aa3c.png)
Simply typing `cmake` or `make` in a terminal with pwd as `aideck-gap8-examples/examples/other/wifi-img-streamer`, didn't work.

**Reply by abyvinodm** [reply]
My environment has libgtk2.0-dev and pkg-config installed.

**Reply by knmcguire** [reply]
Yes that one is a bit confusing, it actually meant the opencv-viewer.py but it is written double.

I'll change it to:
"Clone the [AIdeck example repository](https://github.com/bitcraze/aideck-gap8-examples)

Then run:
```
cd examples/other/wifi-img-streamer
python opencv-viewer.py
```
"

Are you running the script in windows or ubuntu? Have you tried an earlier version of opencv python library?

**Reply by abyvinodm** [reply]
Thank you for clarifying that. Your edits make sense.

I am currently running this script on Ubuntu. I am using opencv-python (4.6.0.66). Is that there a recommended version?

**Reply by knmcguire** [reply]
Hmm that is the very latest version... We do advise to just [install the latest](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/test-functions/wifi-streamer/) but last time we tried it was still the 4.5 version, so there is probably something broken in 4.6

Could you downgrade to the latest 4.5 version? Like this:

`pip install --upgrade opencv-python==4.5.5`

**Reply by abyvinodm** [reply]
Yes, downgrading opencv-python to 4.5.5.64 worked. Thank you.
